src/esp/gfx/Simulator.cpp

```diff
@@ -350,5 +350,19 @@ double Simulator::getWorldTime() {
   return NO_TIME;
 }
 
+bool Simulator::recomputeNavMesh(nav::PathFinder& pathfinder,
+                                 const nav::NavMeshSettings& navMeshSettings) {
+  assets::MeshData::uptr joinedMesh =
+      resourceManager_.createJoinedCollisionMesh(config_.scene.id);
+
+  if (!pathfinder.build(navMeshSettings, *joinedMesh)) {
```

Comment:
This call is going to leak memory here: https://github.com/facebookresearch/habitat-sim/blob/master/src/esp/nav/PathFinder.cpp#L523 and here: https://github.com/facebookresearch/habitat-sim/blob/master/src/esp/nav/PathFinder.cpp#L549